xonsh/tools.py

```diff
@@ -1582,9 +1582,13 @@ def expanduser_abs_path(inp):
 def expand_case_matching(s):
```

Comment:
If you believe so, I can implement using re.match of course! I thought per-character operation was faster but obviously that may not apply to Python as well as it would to C. Going to re-implement using regexps if you all think it's better ^__^